packages/next/src/server-build.ts

```diff
@@ -1131,6 +1131,23 @@ export async function serverBuild({
     });
   }
 
+  if (appPathRoutesManifest) {
+    // create .rsc variant for app lambdas and edge functions
+    // to match prerenders so we can route the same when the
+    // __flight__ header is present
+    const edgeFunctions = middleware.edgeFunctions as Record<string, any>;
```

Comment:
Lets fix the type here so that it always returns `Record<string, EdgeFunction>`

https://github.com/vercel/vercel/blob/7323beea8e52de946756103af76880770717231a/packages/next/src/utils.ts#L2430

Reply:
Fixed in [e941a76](https://github.com/vercel/vercel/pull/8611/commits/e941a762dbc28c44fed4ddc1ac949e7c04c02f21)

Reply:
I might have borked this. Trying again without JS changes, only TS